script/pr-check.js

```diff
@@ -0,0 +1,190 @@
+ */
+function getPRdiff() {
+  return octokit.pulls
+    .get({
+      ...octokitDefaults,
+      mediaType: {
+        format: 'diff',
+      },
+    })
+    .then(({ data }) => data.split('\n'));
+}
+
+/**
+ * A comment is considered outdated if it doesn't have a current `position`.
+ * We also want to ignore comments which weren't made by a user with a matching
+ * login
+ *
+ * @returns {array} An array of objects where each object has the comment `body` along with
+ * the `path` for the file it was left on and the `position` in the file diff
+ */
+function getPRbotComments() {
+  return octokit.paginate(
+    `GET /repos/:owner/:repo/pulls/:pull_number/comments`,
+    octokitDefaults,
+    ({ data }) =>
+      data
+        .filter(
+          comment =>
+            comment.position !== null && comment.user.login === BOT_NAME,
+        )
+        .map(({ path, body, position }) => ({ path, position, body })),
+  );
+}
+
+/**
+ * Create a Github review on the PR, leaving comments if there are any additions made
+ */
+function createReview(additions) {
+  console.log(additions);
+
+  octokit.pulls.createReviewRequest({
+    ...octokitDefaults,
+    team_reviewers: ['frontend-review-group'],
+  });
+
+  return octokit.pulls.createReview({
+    ...octokitDefaults,
+    body: OVERALL_REVIEW_COMMENT,
+    event: 'COMMENT',
+    comments: additions.map(({ path, position }) => ({
+      path,
+      position,
+      body: LINE_COMMENT,
+    })),
+  });
+}
+
+/**
+ * Log the message & data, then exit
+ */
+function finish(message, data) {
+  console.log(message);
+  console.log(data);
+  process.exit();
+}
+
+/**
+ * Add a label to the beginning of added lines in the diff
+ * where the label consists of:
+ *   - filename
+ *   - offset from the top of the diff chunk
+ * And both of the label pieces are separated by a colon
+ *
+ * Modelled after a bash function from this SO answer:
+ * https://stackoverflow.com/a/12179492
+ */
+function labelAdditions(diffLines) {
+  let path = null;
+  let position = null;
+  let inNewFile = false;
+  const output = [];
+
+  diffLines.forEach(line => {
+    if (/--- (a\/)?.*/.test(line)) {
+      return;
+    }
+    const match = line.match(/\+\+\+ (b\/)?(.*$)/);
+    if (match) {
+      path = match[2];
+    } else if (inNewFile && /@@ .* @@.*/.test(line)) {
+      // Only reset the position if our current line is the beginning of a diff chunk
+      // AND if the previous line marked the beginning of a new file
+      position = 1;
+    } else if (/@@ -[0-9]+(,[0-9]+)? \+([0-9]+)(,[0-9]+)? @@.*/.test(line)) {
+      // Increment the position when we reach the beginning of a new diff chunk
+      position++;
+    } else if (/^([ +-]).*/.test(line)) {
+      // Only add to the output if this line of content is an addition (begins with a "+")
+      if (/^[+].*/.test(line)) output.push({ path, position, line });
+      // Increment position for each line of actual content in the diff
+      position++;
```

Comment:
`reduce` might more idiomatic this this need to track / pass accumulators (output, position)

Reply:
I toyed with making it a reducer - I may go back to that.  The tricky part was having it manage the state